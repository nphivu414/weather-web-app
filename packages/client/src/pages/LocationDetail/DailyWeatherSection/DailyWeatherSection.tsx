import * as React from 'react';
import dayjs from 'dayjs';
import { Weather } from 'types';
import { tempFormat } from 'utils';
import { URLS } from 'constants/api';
import { DATE_FORMAT } from 'constants/format';

export type DailyWeatherSectionProps = {
  weatherList: Weather[];
};

const DailyWeatherSection: React.FC<DailyWeatherSectionProps> = ({ weatherList }) => {
  return (
    <div className="p-3">
      <table className="table">
        <tbody>
          {weatherList.map((item) => {
            const { id, applicable_date, min_temp, max_temp, weather_state_abbr } = item;
            const weatherStateImgURL = `${URLS.META_WEATHER_ASSETS}/${weather_state_abbr}.svg`;
            return (
              <tr key={id}>
                <td>{dayjs(applicable_date).format(DATE_FORMAT.DAY_OF_WEEK)}</td>
                <td>{weather_state_abbr && <img className="img" width="24px" src={weatherStateImgURL} />}</td>
                <td className="text-center">
                  <span>{tempFormat(max_temp)}</span>
                </td>
                <td className="text-center">
                  <span className="text-muted">{tempFormat(min_temp)}</span>
                </td>
              </tr>
            );
          })}
        </tbody>
      </table>
    </div>
  );
};

export default DailyWeatherSection;
