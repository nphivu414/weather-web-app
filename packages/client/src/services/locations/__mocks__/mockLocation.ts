import { Place, Weather } from 'types';
export const mockLocations: Place[] = [
  {
    title: 'Helsinki',
    location_type: 'City',
    woeid: 144246,
    latt_long: '13.115,-0.411'
  }
]

export const mockLocationDetail: Place = {
  consolidated_weather: [
    {
      id: '4811453371514880',
      weather_state_name: 'Light Rain',
      weather_state_abbr: 'lr',
      wind_direction_compass: 'S',
      created: '2021-01-05T03:20:16.354553Z',
      applicable_date: '2021-01-04',
      min_temp: 10.335,
      max_temp: 14.04,
      the_temp: 12.84,
      wind_speed: 6.662041035769771,
      wind_direction: 179.13574633334716,
      air_pressure: 1022,
      humidity: 87,
      visibility: 15.07757197963891,
      predictability: 75
    },
  ],
  title: 'San Francisco',
  location_type: 'City',
  woeid: 2487956,
  latt_long: '37.777119, -122.41964',
}

export const mockLocationWeather: Weather = {
  id: '5219640084004864',
  weather_state_name: 'Heavy Cloud',
  weather_state_abbr: 'hc',
  wind_direction_compass: 'NE',
  created: '2021-01-04T09:22:58.583752Z',
  applicable_date: '2021-01-04',
  min_temp: 1.18,
  max_temp: 8.535,
  the_temp: 7.65,
  wind_speed: 8.627662748028845,
  wind_direction: 49.82635538871968,
  air_pressure: 1021.5,
  humidity: 76,
  visibility: 9.605466575200827,
  predictability: 71,
};

export const mockLocationWeatherList: Weather[] = [
  {
    id: '5219640084004864',
    weather_state_name: 'Heavy Cloud',
    weather_state_abbr: 'hc',
    wind_direction_compass: 'NE',
    created: '2021-01-04T09:22:58.583752Z',
    applicable_date: '2021-01-04',
    min_temp: 1.18,
    max_temp: 8.535,
    the_temp: 7.65,
    wind_speed: 8.627662748028845,
    wind_direction: 49.82635538871968,
    air_pressure: 1021.5,
    humidity: 76,
    visibility: 9.605466575200827,
    predictability: 71,
  },
  {
    id: '4892388808982528',
    weather_state_name: 'Heavy Rain',
    weather_state_abbr: 'hr',
    wind_direction_compass: 'SW',
    created: '2021-01-04T12:20:17.765280Z',
    applicable_date: '2021-01-04',
    min_temp: 9.504999999999999,
    max_temp: 12.379999999999999,
    the_temp: 13.325,
    wind_speed: 11.596992554993506,
    wind_direction: 216.04166545749246,
    air_pressure: 1018.5,
    humidity: 92,
    visibility: 8.728090451761712,
    predictability: 77,
  }
]
