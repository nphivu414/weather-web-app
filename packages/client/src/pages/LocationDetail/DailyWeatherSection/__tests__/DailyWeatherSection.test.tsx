import * as React from 'react';
import { shallow } from 'enzyme';

import DailyWeatherSection, { DailyWeatherSectionProps } from '../DailyWeatherSection';
import { mockLocationWeatherList } from 'services/locations/__mocks__/mockLocation';

const props: DailyWeatherSectionProps = {
  weatherList: mockLocationWeatherList,
};

describe('DailyWeatherSection', () => {
  it('should match a snapshot', () => {
    expect(shallow(<DailyWeatherSection {...props} />)).toMatchSnapshot();
  });
});
