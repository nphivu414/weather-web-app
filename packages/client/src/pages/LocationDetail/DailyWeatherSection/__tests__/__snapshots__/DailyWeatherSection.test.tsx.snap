// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyWeatherSection should match a snapshot 1`] = `ShallowWrapper {}`;
